frontend/gateway/grpcclient/client.go

```diff
@@ -386,31 +386,27 @@ func (c *grpcClient) Solve(ctx context.Context, creq client.SolveRequest) (res *
 	if creq.Evaluate {
 		if c.caps.Supports(pb.CapGatewayEvaluateSolve) == nil {
 			req.Evaluate = creq.Evaluate
-		} else {
-			// If evaluate is not supported, fallback to running Stat(".") in order to
-			// trigger an evaluation of the result.
+		} else if c.caps.Supports(pb.CapGatewayEvaluate) == nil {
```

Comment:
This is kind of pointless. There is no combination of buildkit that will support the new cap but didn't support the old cap.

Reply:
Sure, will remove.

The idea was that it would then keep the capabilities as independent, instead of inferring that they must not exist by the non-presence of other capabilities, which would make it easier if we ever decided we wanted to deprecate or change `StatFile` in the future.